modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionServiceTest.java

```diff
@@ -72,140 +105,234 @@ public void tearDown() {
-
-			Assert.fail();
-		}
-		catch (PrincipalException.MustHavePermission principalException) {
-			String message = principalException.getMessage();
-
-			Assert.assertTrue(
-				message.contains(
-					"User " + _user.getUserId() +
-						" must have DELETE permission for"));
-		}
-
-		_testDeleteObjectDefinition(_adminUser, _adminUser);
-		_testDeleteObjectDefinition(_user, _user);
+		ObjectDefinition objectDefinition = _addCustomObjectDefinition(
+			_adminUser);
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustHavePermission.class,
+			StringBundler.concat(
+				"User ", _user1.getUserId(),
+				" must have DELETE permission for ",
+				"com.liferay.object.model.ObjectDefinition ",
+				objectDefinition.getObjectDefinitionId()),
+			() -> _testDeleteObjectDefinition(objectDefinition, _user1));
+
+		_testDeleteObjectDefinition(
+			_addCustomObjectDefinition(_adminUser), _adminUser);
+		_testDeleteObjectDefinition(_addCustomObjectDefinition(_user1), _user1);
 	}
 
 	@Test
 	public void testGetObjectDefinition() throws Exception {
-		try {
-			_testGetObjectDefinition(_adminUser, _user);
-
-			Assert.fail();
-		}
-		catch (PrincipalException.MustHavePermission principalException) {
-			String message = principalException.getMessage();
-
-			Assert.assertTrue(
-				message.contains(
-					"User " + _user.getUserId() +
-						" must have VIEW permission for"));
-		}
-
-		_testGetObjectDefinition(_adminUser, _adminUser);
-		_testGetObjectDefinition(_user, _user);
+		ObjectDefinition objectDefinition = _addCustomObjectDefinition(
+			_adminUser);
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustHavePermission.class,
+			StringBundler.concat(
+				"User ", _user1.getUserId(), " must have VIEW permission for ",
+				"com.liferay.object.model.ObjectDefinition ",
+				objectDefinition.getObjectDefinitionId()),
+			() -> _testGetObjectDefinition(objectDefinition, _user1));
+
+		_testGetObjectDefinition(
+			_addCustomObjectDefinition(_adminUser), _adminUser);
+		_testGetObjectDefinition(_addCustomObjectDefinition(_user1), _user1);
 	}
 
 	@Test
 	public void testPublishCustomObjectDefinition() throws Exception {
-		try {
-			_testPublishCustomObjectDefinition(_user);
-
-			Assert.fail();
-		}
-		catch (PrincipalException.MustHavePermission principalException) {
-			String message = principalException.getMessage();
-
-			Assert.assertTrue(
-				message.contains(
-					"User " + _user.getUserId() +
-						" must have PUBLISH_OBJECT_DEFINITION permission for"));
-		}
-
-		_testPublishCustomObjectDefinition(_adminUser);
+		ObjectDefinition objectDefinition = _addCustomObjectDefinition(
+			_adminUser);
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustHavePermission.class,
+			StringBundler.concat(
+				"User ", _user1.getUserId(),
+				" must have PUBLISH_OBJECT_DEFINITION permission for ",
+				"com.liferay.object "),
+			() -> _testPublishCustomObjectDefinition(objectDefinition, _user1));
```

Comment:
Can we inline the objectDefinition here?

`_testPublishCustomObjectDefinition(_addCustomObjectDefinition(_adminUser), _user1));`

Reply:
if we add `getAdditionalAssertFieldNames`, the tests broken.

Reply:
Inline done!